Sources/libhostmgr/BuildkiteScriptBuilder.swift

```diff
@@ -0,0 +1,165 @@
+import Foundation
+import TSCBasic
+
+public struct BuildkiteScriptBuilder {
+
+    /// A list of files to run `source` against prior to executing the rest of the script.
+    var dependencies = [String]()
+
+    /// A list of key/value pairs representing environment variables that should be defined at the start of the script.
+    var environmentVariables: [String: Value] = [:]
+
+    /// A list of commands to run in the script.
+    var commands = [Command]()
+
+    public init() {
+        self.environmentVariables["BUILDKITE"] = Value(wrapping: "true")
+    }
+
+    /// Add another dependency to the build script.
+    ///
+    /// Each dependency will be placed in a `source $DEPENDENCY` block at the top of the emitted build script
+    public mutating func addDependency(atPath path: String) {
+        self.dependencies.append(path)
+    }
+
+    /// Add an environment variable pair to the build script.
+    ///
+    /// If there's an existing environment variable with the same name, it will be overwritten.
+    public mutating func addEnvironmentVariable(named key: String, value: String) {
+        self.environmentVariables[key] = Value(wrapping: value)
+    }
+
+    /// Copy environment variables from the existing environment into the build script based on their prefix.
+    public mutating func copyEnvironmentVariables(
+        prefixedBy prefix: String,
+        from environment: [String: String] = ProcessInfo.processInfo.environment
+    ) {
+        environment
+            .filter { $0.key.starts(with: prefix) }
+            .forEach { key, value in
+                environmentVariables[key] = Value(wrapping: value)
+            }
```

Comment:
💄  Alternate implementation, that I personally find neater (and also avoids looping twice… even if performance probably doesn't matter here):

```suggestion
        for (key, value) in environment where key.starts(with: prefix) {
            environmentVariables[key] = Value(wrapping: value)
        }
```

Reply:
Love the concise code!